library/spdm_requester_lib/libspdm_req_challenge.c

```diff
@@ -338,18 +338,24 @@ libspdm_return_t libspdm_challenge(void *spdm_context, void *reserved,
 {
     libspdm_context_t *context;
     size_t retry;
+    uint64_t retry_delay_time;
     libspdm_return_t status;
 
     context = spdm_context;
     context->crypto_request = true;
     retry = context->retry_times;
+    retry_delay_time = context->retry_delay_time;
     do {
         status = libspdm_try_challenge(context, slot_id,
                                        measurement_hash_type,
                                        measurement_hash, slot_mask, NULL, NULL, NULL);
         if (LIBSPDM_STATUS_BUSY_PEER != status) {
             return status;
         }
+
+        if (retry_delay_time > 0) {
+            libspdm_sleep_in_us(retry_delay_time);
```

Comment:
do we need such check `retry_delay_time > 0` ?

I think libspdm_sleep_in_us(0) should be OK ?

Reply:
It's just about the concern that sleep(0) would yield control back to the scheduler and lead to unexpected result.

Reply:
Good point. (https://learn.microsoft.com/en-us/windows/win32/api/synchapi/nf-synchapi-sleep)

I think we should document the expected behavior for 0 anyway. (https://github.com/DMTF/libspdm/blob/main/include/hal/library/platform_lib.h#L27)

Reply:
I will add appropriate notation.